heed/src/cookbook.rs

```diff
@@ -0,0 +1,450 @@
+//! A cookbook of examples on how to use heed. Here is the list of the different topics you can learn about:
+//!
+//! - [Listing and Opening the Named Databases](#listing-and-opening-the-named-databases)
+//! - [Create Custom and Prefix Codecs](#create-custom-and-prefix-codecs)
+//! - [Change the Environment Size Dynamically](#change-the-environment-size-dynamically)
+//! - [Decode Values on Demand](#decode-values-on-demand)
+//! - [Advanced Multithreaded Access of Entries](#advanced-multithreaded-access-of-entries)
+//!
+//! # Listing and Opening the Named Databases
+//!
+//! Sometimes it is useful to list the databases available in an environment.
+//! LMDB stores them already in the unnamed database, a database that is always
+//! there in which you can write.
+//!
+//! However, once you create new databases, after defining the [`EnvOpenOptions::max_dbs`]
+//! parameter, the names of those databases are automatically stored in the unnamed one.
+//!
+//! ```
+//! use std::error::Error;
+//! use std::fs;
+//! use std::path::Path;
+//!
+//! use heed::types::*;
+//! use heed::{Database, EnvOpenOptions};
+//!
+//! fn main() -> Result<(), Box<dyn Error>> {
+//!     let env_path = Path::new("target").join("heed.mdb");
+//!
+//!     fs::create_dir_all(&env_path)?;
+//!
+//!     let env = unsafe {
+//!         EnvOpenOptions::new()
+//!             .map_size(10 * 1024 * 1024) // 10MB
+//!             .max_dbs(3) // Number of opened databases
+//!             .open(env_path)?
+//!     };
+//!
+//!     let rtxn = env.read_txn()?;
+//!     let unnamed: Database<Str, DecodeIgnore> =
```

Comment:
Maybe we could put a small comment explaining that we don’t know the value part of the database? (is that even true? Wasn’t the value just empty or `()` in rust?)